aspnetcore/blazor/host-and-deploy/webassembly.md

```diff
@@ -47,7 +47,14 @@ dotnet publish -c Release
 
 WebAssembly AOT compilation is only performed when the project is published. AOT compilation isn't used when the project is run during development (`Development` environment) because AOT compilation usually takes several minutes on small projects and potentially much longer for larger projects. Reducing the build time for AOT compilation is under development for future releases of ASP.NET Core.
 
-The size of an AOT-compiled Blazor WebAssembly app is generally larger than the size of the app if compiled into .NET IL. Although the size difference depends on the app, most AOT-compiled apps are about twice the size of their IL-compiled versions. This means that using AOT compilation trades off load time performance for runtime performance. Whether this tradeoff is worth using AOT compilation depends on your app. Blazor WebAssembly apps that are CPU intensive generally benefit the most from AOT compilation.
+The size of an AOT-compiled Blazor WebAssembly app is generally larger than the size of the app if compiled into .NET IL:
+
+* Although the size difference depends on the app, most AOT-compiled apps are about twice the size of their IL-compiled versions. This means that using AOT compilation trades off load-time performance for runtime performance. Whether this tradeoff is worth using AOT compilation depends on your app. Blazor WebAssembly apps that are CPU intensive generally benefit the most from AOT compilation.
+
+* The larger size of an AOT-compiled app is due to two conditions:
+
+  * More code is required to represent high-level .NET IL instructions in native WebAssembly.
+  * AOT does ***not*** trim out managed DLLs when the app is published. Blazor must fall back to using the .NET interpreter with the .NET binaries for JSON serialization and deserialization, which use reflection and dynamic code generation. Requiring the DLLs on the client increases the download size but provides a more compatible .NET experience.
```

Comment:
More generally, the DLLs are needed for reflection metadata and to enable certain .NET runtime features like reflection handling. 

@lewing  How does this sound?

```suggestion
  * AOT does ***not*** trim out managed DLLs when the app is published. Blazor still needs the DLLs for reflection metadata and to support certain .NET runtime features. Requiring the DLLs on the client increases the download size but provides a more compatible .NET experience.
```

Reply:
If we can say what ...

> certain .NET runtime features

... are, I think it would be helpful/interesting/clear. I think the reader hits this and stops to think, "Why can't they just tell me?"